otel-tests/src/lib.rs

```diff
@@ -0,0 +1,69 @@
+use proc_macro::TokenStream;
+use quote::quote;
+use quote::ToTokens;
+use syn::ItemFn;
+use syn::Lit;
+use syn::MetaNameValue;
+use syn::Path;
+
+#[proc_macro_attribute]
+pub fn otel_test(attribute: TokenStream, item: TokenStream) -> TokenStream {
+    let fn_item = syn::parse::<ItemFn>(item).unwrap();
+
+    let sig = fn_item.sig;
+    let block = fn_item.block;
+    let name = &sig.ident;
+    let attrs = fn_item.attrs;
+
+    let test = if attrs.is_empty() {
+        quote!(#[tokio::test])
+    } else {
+        quote!()
+    };
+
+    let init_tracing = if attribute.is_empty() {
+        quote!(::daemon_tests::init_tracing)
+    } else {
+        let attribute = syn::parse::<MetaNameValue>(attribute).unwrap();
+        if !attribute.path.is_ident("init") {
+            panic!(
+                "Unexpected attribute {:?}",
+                attribute.path.get_ident().unwrap()
+            )
+        }
+
+        let path = match attribute.lit {
+            Lit::Str(lit) => lit.parse::<Path>().expect("Expected function path"),
+            other => panic!(
+                "Expected string function path, got {}",
+                other.to_token_stream()
+            ),
+        };
+
+        quote!(#path)
+    };
+
+    let tokens = quote! {
+        #test
+        #(#attrs)*
+        #sig {
+            #init_tracing();
+
+            let caught = {
+                futures::FutureExt::catch_unwind(std::panic::AssertUnwindSafe(tracing::Instrument::instrument(
+                    async #block, tracing::info_span!(stringify!(#name))
+                ))).await
+            };
+
+            #[allow(clippy::disallowed_methods)]
+            tokio::time::sleep(std::time::Duration::from_secs(1)).await;
+            opentelemetry::global::force_flush_tracer_provider();
+
+            if let Err(e) = caught {
+                panic!("{:?}", e);
```

Comment:
nit: I believe you can use short-hand notation:
```suggestion
                panic!("{e:?}");
```

also: wouldn't `e:#` be better here?

Reply:
I tried `panic!("{e:#?}");`, but it gave this error:
```
    = note: did you intend to capture a variable `e` from the surrounding scope?
    = note: to avoid ambiguity, `format_args!` cannot capture variables when the format string is expanded from a macro
    = note: this error originates in the attribute macro `otel_test` (in Nightly builds, run with -Z macro-backtrace for more info)
```

Reply:
oh wow, I've learned something new! to be fair, I've never tried this is a macro before ;)